lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -513,6 +523,39 @@ fromNetworkMagic :: NetworkMagic -> W.ProtocolMagic
 fromNetworkMagic (NetworkMagic magic) =
     W.ProtocolMagic (fromIntegral magic)
 
+--
+-- Stake pools
+--
+
+fromPoolId :: SL.KeyHash 'SL.StakePool crypto -> W.PoolId
+fromPoolId (SL.KeyHash x) = W.PoolId $ getHash x
+
+fromPoolDistr
+    :: SL.PoolDistr TPraosStandardCrypto
+    -> Map W.PoolId Percentage
+fromPoolDistr =
+    Map.map (unsafeMkPercentage . fst)
+    . Map.mapKeys fromPoolId
+    . SL.unPoolDistr
+
+-- TODO: Change to return a map of maps, instead of using head
+fromRewards
+    :: O.NonMyopicMemberRewards TPraosStandardCrypto
+    -> Map W.PoolId (Quantity "lovelace" Word64)
+fromRewards =
+    Map.map (Quantity . fromIntegral)
+    . Map.mapKeys fromPoolId
+    . snd
+    . head
```

Comment:
Not sure what we want to do here. We are currently only requesting rewards for one wallet at a time, so in a way perhaps we should keep it (albite with a better error than `head`)

Reply:
I agree that having `head` _here_ feels kinda wrong. If anything, the caller of `fromRewards` should make the decision about how to treat the result but this function `fromRewards` should simply do the type translation, nothing more.

Reply:
b36cfe6bebe49b961356d60cd36196b47a9b901e
(But still treating this as an `error`)